packages/docusaurus-theme-redoc/src/types/modules.ts

```diff
@@ -1,19 +1,50 @@
 interface SpecProps {
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  /**
+   * Spec to use, already loaded previously
+   */
   spec: import('redoc/typings/types').OpenAPISpec;
-  url?: string;
-  isSpecFile?: boolean;
+  /**
+   * When spec not provided, load the spec from docusaurus config
+   * fallback to first configuration if not provided
+   */
+  id?: string;
+  /**
+   * docusaurus theme to use
+   */
   themeId?: string;
 }
+interface SpecResult {
+  hasLogo: boolean;
+  spec: import('redoc/typings/types').OpenAPISpec;
+  store: import('redoc/typings').AppStore;
+  options: import('redoc/typings').RedocRawOptions;
+  darkThemeOptions: import('redoc/typings').RedocRawOptions;
+  lightThemeOptions: import('redoc/typings').RedocRawOptions;
+}
+
+type RedocProps = SpecProps & {
```

Comment:
this is what I was talking about here - https://github.com/rohit-gohri/redocusaurus/pull/339/files#r1518118853

The id is now available as a prop in RedocProps also though SpecProps. But it doesn't work

Reply:
Hmm strange, when I run the website it looks ok.

But I can be wrong as every OpenAPI example in the website contains approximative content. We should make many OpenAPI example and replace Dog, Cat ... with new animals (Elephant...) to be sure the right OpenAPI is loaded

Reply:
It looks okay, but this compnent now has props which don't do anything

Reply:
Are you sure it's not working with `id` ? Everything is given to `useSpec` so it should works

Should be added there in that case https://github.com/rohit-gohri/redocusaurus/pull/339/files#diff-28d0aac1a088788b029652a42c7bfb38079857218f6f9f29be02cf9dff768a2cR43